### PR TITLE
[ng] examples(toh-6, server-communication): switch to using the stream_transform pkg

### LINKED ISSUE #709: [ng] examples(toh-6, server-communication): switch to using the stream_transform pkg
Currently the `toh-6` and `server-communication` examples use https://pub.dartlang.org/packages/stream_transformers but it is not strong mode compliant (see #703).

Switch to using https://pub.dartlang.org/packages/stream_transform.

cc @kwalrath

### PR DESCRIPTION
**Pull request by chalin**
Fixes #709.

Note that after this fix:
- toh-6 still fails to compile under dartdevc due to the errors noted below.
- server-communication is free from ddc compiler errors.

Toh-6 ddc errors:
```
Error compiling dartdevc module:angular_tour_of_heroes|lib/lib__hero_detail_component.template.js

[error] Element 'Location' from SDK library 'dart:html' is implicitly hidden by 'location.dart'. (package:angular_tour_of_heroes/hero_detail_component.template.dart, line 259, col 84)
[error] Element 'Location' from SDK library 'dart:html' is implicitly hidden by 'location.dart'. (package:angular_tour_of_heroes/hero_detail_component.template.dart, line 260, col 54)
```

### CODE REVIEW

File: examples/ng/doc/toh-6/test/hero_search.dart [hunk 1]
```diff
@@ -95,10 +95,12 @@ void heroSearchTests() {
 
 Future _typeSearchTextAndRefreshPO(String searchText) async {
   await po.search.type(searchText);
-  Stream<List> herosStream;
-  await fixture.update((c) => herosStream = c.heroes);
-  // Wait for the data to be fetched from db.
-  // (We could set a timeout, but it isn't needed with the mock db.)
-  await herosStream.first;
+  // Waiting on the stream doesn't work anymore. Will just use a timer for now:
```
Comment (kwalrath) [inline examples/ng/doc/toh-6/test/hero_search.dart]: Why not?

Comment (chalin) [inline examples/ng/doc/toh-6/test/hero_search.dart]: https://github.com/dart-lang/stream_transform/issues/18

I'll update the text to refer to this issue.

**Comment by chalin** [comment]
Also see https://github.com/dart-lang/stream_transform/issues/18.

**Comment by chalin** [comment]
This has been approved, and Travis was green across the board. The last commit essentially just tweaked comments, so merging now.

**Pull request merged**